tests/service_info.py

```diff
@@ -0,0 +1,32 @@
+"""Service info test helpers."""
+
+from __future__ import annotations
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigFlowResult
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.service_info.dhcp import DhcpServiceInfo
+
+
+class MockDhcpServiceInfo(DhcpServiceInfo):
+    """Mocked DHCP service info."""
+
+    def __init__(self, ip: str, hostname: str, macaddress: str) -> None:
+        """Initialize the mock service info."""
+        # Historically, the MAC address was formatted without colons
+        # and since all consumers of this data are expecting it to be
+        # formatted without colons we will continue to do so
+        super().__init__(
+            ip=ip,
+            hostname=hostname,
+            macaddress=dr.format_mac(macaddress).replace(":", ""),
+        )
+
+    async def start_discovery_flow(
+        self, hass: HomeAssistant, domain: str
+    ) -> ConfigFlowResult:
+        """Start a reauthentication flow."""
+        return await hass.config_entries.flow.async_init(
+            domain, context={"source": config_entries.SOURCE_DHCP}, data=self
+        )
```

Comment:
I'm not a fan of this one.

This is an essential and common part of tests, that is now hidden away more without needing it (IMHO).

In general, we have been moving away from helpers around these things (including for example service and state test helpers).

I feel like this is going in the opposite direction.

The gain is is mostly the this bit:

```
    result = await hass.config_entries.flow.async_init(
        DOMAIN, context={"source": SOURCE_DHCP}, data=info
    )
```

Which in this example I personally would consider a lot, as this is how we init config flows for all other things. With the suggested change this all of a sudden becomes different for this specific case, making things less consistent.

../Frenck

Reply:
The original purpose was to have a helper for "format mac as dhcp discovery does it" as a lot of the current tests differ from real life scenarios 

I felt having the start discovery helper aligned nicely with start reauth and start reconfigure flows from a mock config entry but I can remove it

Reply:
I have simplified the PR...
I hope that I understood your comment correctly